NuRadioReco/modules/io/coreas/readCoREASStationGrid.py

```diff
@@ -0,0 +1,201 @@
+    This module distributes core positions randomly within a user defined area and calculates the electric field
+    at the detector positions as specified in the detector description by choosing the closest antenna 
+    of the star shape pattern simulation
+    """
+
+    def __init__(self):
+        self.__t = 0
+        self.__t_event_structure = 0
+        self.__t_per_event = 0
+        self.__input_files = None
+        self.__station_id = None
+        self.__n_cores = None
+        self.__max_distace = None
+        self.__current_input_file = None
+        self.__random_generator = None
+        self.logger = logging.getLogger('NuRadioReco.readCoREAS')
+
+    def begin(self, input_files, xmin, xmax, ymin, ymax, n_cores=10, seed=None, log_level=logging.INFO):
+        """
+        begin method
+
+        initialize readCoREAS module
+
+        Parameters
+        ----------
+        input_files: input files
+            list of coreas hdf5 files
+        xmin: float
+            minimum x coordinate of the area in which core positions are distributed
+        xmax: float
+            maximum x coordinate of the area in which core positions are distributed
+        ymin: float
+            minimum y coordinate of the area in which core positions are distributed
+        ynax: float
+            maximum y coordinate of the area in which core positions are distributed
+        n_cores: number of cores (integer)
+            the number of random core positions to generate for each input file
+        seed: int (default: None)
+            Seed for the random number generation. If None is passed, no seed is set
+        """
+        self.__input_files = input_files
+        self.__n_cores = n_cores
+        self.__current_input_file = 0
+        self.__area = [xmin, xmax, ymin, ymax]
+
+        self.__random_generator = numpy.random.RandomState(seed)
+        self.logger.setLevel(log_level)
+
+    @register_run()
+    def run(self, detector, output_mode=0):
+        """
+        Read in a random sample of stations from a CoREAS file.
+        For each position the closest observer is selected and a simulated
+        event is created for that observer.
+
+        Parameters
+        ----------
+        detector: Detector object
+            Detector description of the detector that shall be simulated
+        output_mode: integer (default 0)
+            0: only the event object is returned
+            1: the function reuturns the event object, the current inputfilename, the distance between the choosen station and the requested core position,
+               and the area in which the core positions are randomly distributed
+
+
+        """
+        while (self.__current_input_file < len(self.__input_files)):
+            t = time.time()
+            t_per_event = time.time()
+            filesize = os.path.getsize(self.__input_files[self.__current_input_file])
+            if(filesize < 18456 * 2):
```

Comment:
Where does this specific value for the filesize cut come from?

Reply:
the number is based on observations. Correct files will always be larger.

Reply:
Could you add a comment please? It looks pretty specific, so it's a bit confusing